### DISCUSSION TITLE
Episode: Juri NX

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#69] [Announcements]
## Background:

https://nx.dev

## Intro

Andrew: Hello, welcome to the devtools.fm podcast. This is a podcast about developer tools and the people who make them. I'm Andrew and this is my co-host Justin. 

Justin: Hey everyone! Our guest today is Juri Strompflohner. Juri is the director of developer experience at nrwl.io and is joining us today to talk about Nx. Juri, would you like to tell our listeners a little more about yourself?

## Questions

- [x] What is Nx?
- [x] We love monorepo, but for our listeners that might not know what a monorepo is could you tell use what it is and the benefits of using one?
- [x] How does Nx work? How is it different from something like lerna?
  - [x] how do builds work? does nx help me build for cjs/esm/web targets?
- [x] What are some of the cool things that NX does that make it run so fast?
  - [x] When does Nx do "Distributed Task Execution"? Why might one want to run code like that?
- [x] Nx works by caching build outputs, how much does it cache? Is the cache distributed? If so what does that mean for devs?
- [x] Does NX only support javascript? Can it be used to run a polyglot monorepo?
- [x] Lerna is really good at publishing packages for monorepos. Does NX help you with the publishing part? How does publishing work for polyglot repos?
- [ ] ~What is the "Nx Daemon"? How does it help to speed up development?~
- [x] What does an NX plugin do? What is your favorite plugin?

## Tooltips

Add them in the comments below!

## Outro

Andrew: Well that’s it for this week’s episode of devtools.fm. Be sure to follow us on YouTube and wherever you consume your podcasts. Thanks for listening!

**Comment by hipstersmoothie** [comment]
Andrew Tooltips:

- [Vite Virtual Modules](https://vitejs.dev/guide/api-plugin.html#virtual-modules-convention)
- [swc](https://swc.rs)

**Comment by just-be-dev** [comment]
- https://github.com/yjs/yjs
- https://github.com/jdoleary/WebsocketPie

**Comment by juristr** [comment]
- [Nx Console](https://marketplace.visualstudio.com/items?itemName=nrwl.angular-console)
- https://obsidian.md/
- [LunarVim](https://www.lunarvim.org/#opinionated)

**Comment by just-be-dev** [comment]
## Show notes

https://monorepo.tools/
https://nx.dev/packages/react
https://marketplace.visualstudio.com/items?itemName=nrwl.angular-console
https://nx.dev/guides/nx-devkit-angular-devkit#nx-devkit-and-angular-devkit
https://github.com/phenomnomnominal/tsquery
https://nx.dev/structure/buildable-and-publishable-libraries
